test/Jsr292/src/com/ibm/j9/jsr292/indyn/IndyTest.java

```diff
@@ -211,17 +211,21 @@ public void test_indyn_call_Illegal_crosspackage_method () {
 	//Negative test : invalid bootstrap method
 	@Test(groups = { "level.extended" })
 	public void test_indyn_invalid_bootstrap_method() {
-		
-		boolean bootstrapMethodError = false;
-		
 		try {
 			String s = com.ibm.j9.jsr292.indyn.GenIndyn.invalid_bootstrap(new Object());
-		} catch ( BootstrapMethodError e ) {
-			bootstrapMethodError = true;
-		}
-		
-		if ( bootstrapMethodError == false ) {
-			Assert.fail("BootstrapMethodError not thrown when invalid bootstrap method is used");
+			Assert.fail("Exception should be thrown");
+		} catch (BootstrapMethodError e) {
+			//BootstrapMethodError should only apply to Java8
+			if (false == "1.8".equals(System.getProperty("java.specification.version"))) {
+				Assert.fail("Unexpected exception ", e);
+			}
+		} catch (NoSuchMethodError e) {
+			//Java9 and up throws NoSuchMethodError
+			if ("1.8".equals(System.getProperty("java.specification.version"))) {
+				Assert.fail("Unexpected exception ", e);
+			}
+		} catch (Exception e) {
+			Assert.fail("Unexpected exception ", e);
```

Comment:
if there is no exception thrown, then this test will pass. we want this scenario to fail. so, you will need to add an `Assert.fail` outside the `catch` block.

Reply:
The test will fail if no exception is thrown, as 
```
 			String s = com.ibm.j9.jsr292.indyn.GenIndyn.invalid_bootstrap(new Object());
 -		} catch ( BootstrapMethodError e ) {		 +			Assert.fail("Exception should be thrown");
```
adding extra assert check outside it redundant.
catching all exception and fail it with the assert API makes testng easy to control the error output.

Reply:
correct - `Assert.fail("Exception should be thrown");`

Reply:
I think we can be more clear on the error msg. Instead of `Exception should be thrown`, we can use Babneet's suggestion
`BootstrapMethodError (in Java 8) or NoSuchMethodError (Java 9+) should have been thrown`

Reply:
great, PR updated,